django_mongodb/compiler.py

```diff
@@ -591,42 +603,44 @@ def execute_sql(self, result_type):
                         f"{field.__class__.__name__}."
                     )
             prepared = field.get_db_prep_save(value, connection=self.connection)
-            values.append((field, prepared))
+            if hasattr(value, "as_mql"):
+                prepared = prepared.as_mql(self, self.connection)
+            values.append((field.column, prepared))
+        try:
+            criteria = self.build_query().mongo_query
+        except EmptyResultSet:
+            return 0
         is_empty = not bool(values)
-        rows = 0 if is_empty else self.update(values)
+        if is_empty:
+            rows = 0
+        else:
+            base_pipeline = [
```

Comment:
This part is the only thing that need focus. Here I just create the **$merge** pipeline and the **affected rows** pipeline with a single transaction. I really don't know if this is the way to go.

Reply:
This is the right way to go about it. having a transaction ensures that state isn't changed between the lookup and the update.

Reply:
I think we can use update many directly as Shane mentioned. If I've re-read the docs  and what the docs support are this three stages.

Reply:
Yeap, It works 🚀. So we are able to do the update without making two queries (one for count and the other for the update)